src/lib.rs

```diff
@@ -331,6 +336,45 @@ impl PartialEq<[u8]> for Hash {
 
 impl Eq for Hash {}
 
+#[cfg(feature = "rkyv")]
+/// This implementation is constant-time.
+impl PartialEq for ArchivedHash {
+    #[inline]
+    fn eq(&self, other: &ArchivedHash) -> bool {
+        constant_time_eq::constant_time_eq_32(&self.0, &other.0)
+    }
+}
+
+#[cfg(feature = "rkyv")]
+/// This implementation is constant-time.
+impl PartialEq<[u8; OUT_LEN]> for ArchivedHash {
+    #[inline]
+    fn eq(&self, other: &[u8; OUT_LEN]) -> bool {
+        constant_time_eq::constant_time_eq_32(&self.0, other)
+    }
+}
+
+#[cfg(feature = "rkyv")]
+/// This implementation is constant-time if the target is 32 bytes long.
+impl PartialEq<[u8]> for ArchivedHash {
+    #[inline]
+    fn eq(&self, other: &[u8]) -> bool {
+        constant_time_eq::constant_time_eq(&self.0, other)
+    }
+}
+
+#[cfg(feature = "rkyv")]
+/// This implementation is constant-time if the target is 32 bytes long.
+impl PartialEq<Hash> for ArchivedHash {
+    #[inline]
+    fn eq(&self, other: &Hash) -> bool {
+        constant_time_eq::constant_time_eq(&self.0, &other.0)
+    }
+}
+
+#[cfg(feature = "rkyv")]
+impl Eq for ArchivedHash {}
+
 impl fmt::Display for Hash {
```

Comment:
Shouldn't `ArchivedHash` implement `Display` and `Debug` too?  Also, `Hash` derives `std::hash::Hash`.  Might be useful as well.